### DISCUSSION TITLE
Gitlab Integration: Add support for Masked and hidden variables

### DISCUSSION THREAD

**Discussion by GameBurrow** [Infisical/infisical#6312] [Feature Request]
### Feature description
Recently Gitlab has added new variable visiblity type:
![image](https://github.com/user-attachments/assets/9cea48cc-d881-4c2c-b5ee-85423c9578f0)

Would be nice to have this option support also in infisical gitlab integration.

### Why would it be useful?
Prevents displaying Infisical secrest completely in Gitlab for improving security.

**Comment by akhilmhdh** [comment]
Moving this to [feature request discussion](https://github.com/Infisical/infisical/issues/5172).
